modules/ROOT/pages/parallel-foreach-scope.adoc

```diff
@@ -47,7 +47,7 @@ Every execution of the Parallel ForEach scope will start with the same variables
 ----
```

Comment:
HIDDEN TEXT 
**edits in bold**:

= Parallel For Each Scope       **>  =Configure the Parallel for Each Scope Feature**
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: anypoint studio, studio, mule, split, aggregate, scope, parallel, for, each
[IMPORTANT]            ** >  Replace this Important notice with an introduction text arranged as follows:
What the feature is and why your audience wants to use it
Not available in the Anypoint Studio *Mule Palette* view and so must be configured**

This component is not available in the Anypoint Studio *Mule Palette*. You are able to configure and use the Parallel For Each scope by manually modifying the XML configuration.         **> Move this as indicated in the following edit suggestion.**

The Parallel For Each scope splits a collection and parallel processes each part of the collection with the components within the scope (subject to any limitation in concurrency that has been configured). The output is a collection with the result of each route. After all messages are processed, the results are aggregated before continuing within the flow in the same order they were in prior to the split.  
          **>  I am guessing here, so please read the text that I am suggesting very carefully (in future, I will probably return such requests to you, so that you can make a first pass at rewording them before I try to do it, since I am not the expert that you are)....**
         **> The Parallel For Each Scope feature enables you to <more quickly and efficiently> process a collection of messages by splitting the collection into parts that can be separately but simultaneously processed within the scope of existing concurrent-processing restrictions. After all messages are processed, the results are aggregated, the collection reassembled, and the flow continued, in much less time than linear processing takes.** 
       **> However, because this feature is not available in the Anypoint Studio *Mule Palette* view, you must manually configure Parallel For Each Scope in the XML, as follows:**

Reply:
Hidden Line 71:
Does the audience understand what it means/why it is important to note that none of the modifications is registered? Is this an advantage or a disadvantage?

Reply:
HIDDEN TEXT Lines 73-75
**edits in bold**

== Error Handling
Every route is processed in parallel. So, when an error is thrown in one route, Parallel For Each continues to process each route and then invokes the xref:intro-error-handlers.adoc[Error Handler] with the aggregation of the results and errors of all the routes, for example:
**> The previous text does not refer to the individual messages or "parts of the collection" as "routes": should it? Seems we ought to introduce and use this term consistently.**
**>  == Error Handling
Because every route is processed in parallel, if an error is thrown in one route, processing continues in all of the other routes until all results (and any errors) can be aggregated and then processed by the xref:intro-error-handlers.adoc[Error Handler], as shown here:**

Reply:
HIDDEN TEXT Lines 103-7


=== Throws
* `MULE:COMPOSITE_ROUTING`
include::partial$parallelforeach_vs_foreach.adoc[]

**> I have no idea how this is a subtopic of Error Handling or what is trying to be conveyed in this subtopic. This needs to be reworked and clarified.**